### PR TITLE
Added samples for the Git Commits endpoint.

### PR DESCRIPTION
**Pull request by shadyysf**

### CODE REVIEW

File: ClientLibrary/Snippets/Microsoft.TeamServices.Samples.Client/Git/CommitsSample.cs [hunk 1]
```diff
@@ -0,0 +1,141 @@
+﻿using System.Collections.Generic;
+using Microsoft.TeamFoundation.SourceControl.WebApi;
+
+namespace Microsoft.TeamServices.Samples.Client.Git
+{
+    [ClientSample(GitWebApiConstants.AreaName, "commits")]
```
Comment (willsmythe) [inline ClientLibrary/Snippets/Microsoft.TeamServices.Samples.Client/Git/CommitsSample.cs]: No constant for "commits" in the public client library? If not, you should log an internal work item to address this at some point and then loop back and fix the sample. I'd expect a constant in `GitWebApiConstants`

File: ClientLibrary/Snippets/Microsoft.TeamServices.Samples.Client/Git/CommitsSample.cs [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+        [ClientSampleMethod]
+        public List<GitCommitRef> GetCommitsInDateRange()
+        {
+            GitRepository repo = GitSampleHelpers.FindAnyRepositoryOnAnyProject(this.Context);
+
+            return this.Context.Connection.GetClient<GitHttpClient>()
+                .GetCommitsAsync(repo.Id, new GitQueryCommitsCriteria()
+                {
+                    FromDate = "6/14/2018 6:19",
```
Comment (willsmythe) [inline ClientLibrary/Snippets/Microsoft.TeamServices.Samples.Client/Git/CommitsSample.cs]: It's a little odd that `GitQueryCommitsCriteria` uses `String` for FromDate and ToDat. I would consider having the sample  show how to construct a `DateTime` from a string (assuming this is how you'd normally create a DateTime from some string date value) and then convert this into a standard ISO 8601 formatted date/time value, which presumably the server accepts/understands. Feels more deterministic and gives the client full control.

File: ClientLibrary/Snippets/Microsoft.TeamServices.Samples.Client/Git/CommitsSample.cs [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    [ClientSample(GitWebApiConstants.AreaName, "commits")]
+    public class CommitsSample : ClientSample
+    {
+        [ClientSampleMethod]
+        public List<GitCommitRef> GetAllCommits()
+        {
+            GitRepository repo = GitSampleHelpers.FindAnyRepositoryOnAnyProject(this.Context);
+
+            return this.Context.Connection.GetClient<GitHttpClient>()
+                .GetCommitsAsync(repo.Id, new GitQueryCommitsCriteria()).Result;
```
Comment (alepauly) [inline ClientLibrary/Snippets/Microsoft.TeamServices.Samples.Client/Git/CommitsSample.cs]: Ideally we should use project.Id, repo.Id to force the Urls used to be project scoped.

Comment (alepauly) [reply ClientLibrary/Snippets/Microsoft.TeamServices.Samples.Client/Git/CommitsSample.cs]: I'll be changing a bunch to this as I work through them so you can leave this one for now if you want.

Comment (shadyysf) [reply ClientLibrary/Snippets/Microsoft.TeamServices.Samples.Client/Git/CommitsSample.cs]: Sounds good.

**Pull request merged**